Speed up object initialization in AArch64 using SIMD instructions

nrainhart opened the pull request:
This PR adds initial support for SIMD instructions targeting the AArch64 architecture, and uses them to speed up object initialization (on JIT-ed `new` primitives) by ~2x, while still supporting the previous scalar implementation on other platforms.

Commit:
Allow testing machine instruction generation without Unicorn

Commit:
Allow encoding DUP machine instruction

Commit:
Allow encoding ST1 machine instruction

Commit:
A

Commit:
Use new SIMD machine instructions to initialize new objects

Commit:
Cleanup

Commit:
Use SIMD instructions to initialize dynamically-sized objects

Commit:
Add fallback when arm64 SIMD instructions are not available

Commit:
Fix comment

Commit:
Test new fixed-size objects are initialized correctly

Commit:
Fix scalar genPrimitiveNew

Commit:
Test new variable-size objects are initialized correctly

Commit:
Name vectror registers

nrainhart commented on smalltalksrc/VMMakerTests/VMARMV8SIMDEncodingTest.class.st:
```diff
@@ -0,0 +1,65 @@
+Class {
```
These tests were written while incrementally adding SIMD instructions to Pharo, but I'm not sure how valuable it is to keep them, considering that:
- these tests are pretty platform-specific (we disassemble the instructions and check that they're the expected AArch64 SIMD instruction)
- we're also exercising the actual instruction behavior when testing the primitives that use the SIMD instructions

If we removed these tests, we wouldn't need the `DummyProcessor` we're also adding as part of this PR.

nrainhart replied:
Actually... the `DummyProcessor` was needed since the `UnicornProcessor` is not working properly in my Mac with M1. So the discussion on whether to keep it or not should be based on whether we consider it worth it to keep a stubbed processor.

guillep commented:
This requires a bit more of work. Latest changes do not compile.
Delegating to the backend was transpiled to the following:

```c
fillFromuntilwithusingVr(backend(), Arg1Reg, byteSizeReg, fillReg, VReg1);
```

but: 
 - backend does not exist of course :)
 - the vector registers have not been generated

Commit:
Fix VM build error

Commit:
fix inline with backend variable

Slang incorrectly translates the backend variable into a self_in_xxx_backend.
This is solved by moving the usage of the variable inside the Cogit class (and respecting demeter's law)

Commit:
Avoid generating inlined methods with same name

Commit:
Tweak method for Slang to generate the correct C output

nrainhart commented:
@guillep fixed the transpilation errors. This should be ready for a second pass!

tesonep committed:
Merge branch 'pharoX' into initialization-fill-intrinsic
